### PR TITLE
GEODE-47: Add CLI option to run locators in foreground

### PR DESCRIPTION
**Pull request by balesh2**
* add a command line option to start locators in the foreground. The gfsh command will not return until the locator stops running.

* add unit tests for new CLI option

* add distributed unit tests for new CLI option

* add unit test for new Builder options

* add CLI Strings for foreground option

* add jmh compile dependency to fix intelliJ build in case it isn't inherited from main

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by mstolz1** [comment]
My preference would be to NOT pollute stdout with all the log data and keep
the logs going to the file they normally would. Otherwise, I will end up
having to redirect stdout to somewhere and can't see what it says without
tailing that file. It would also potentially make it more difficult to
actually gather logs and stats files via the gfsh command for doing so
because we won't know where they are.

--
Mike Stolz
Principal Engineer, GemFire Product Lead
Mobile: +1-631-835-4771

**Comment by metatype** [comment]
I agree with @jdeppe-pivotal that a process started in the foreground should be logging to stdout.  The intent of the command is to make it easier to run geode in containers and allow the "platform" to collect and manage logs.  For example, see the `docker logs` command:  https://docs.docker.com/engine/reference/commandline/logs/#usage.

Adding to the Jen's comment on `ctrl-c`, the process to be stopped is the geode locator or server, not the `gfsh` process.

**Comment by balesh2** [comment]
Closing this until we have further discussions about the requirements and implementation

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/StartLocatorCommand.java [hunk 1]
```diff
@@ -350,6 +355,26 @@ ResultModel doStartLocator(
...
+        if (Thread.interrupted()) {
+          return result;
+        }
+
+        locatorState = LocatorLauncher.LocatorState.fromDirectory(workingDirectory, memberName);
+        Gfsh.print("\rUptime: " + locatorState.getUpTimeAsFormattedString());
+        synchronized (this) {
+          TimeUnit.MILLISECONDS.timedWait(this, 1000);
+        }
+      } while (locatorState.getStatus().equals(AbstractLauncher.Status.ONLINE));
```
[the remark this answers is not in the document]
Comment (balesh2) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/StartLocatorCommand.java]: That's a good point. If this is the implementation we go with, I will be sure to handle that case here.

[the remark this answers is not in the document]
Comment (onichols-pivotal) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/StartLocatorCommand.java]: while (state==STARTING || state==ONLINE) should be sufficient

**Pull request closed without merging**